### PR TITLE
Add various tests.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

File: test/js-api/global/constructor.any.js [hunk 1]
```diff
@@ -118,15 +117,45 @@ for (const type of ["i32", "f32", "f64"]) {
...
+  test(() => {
+    const argument = { "value": type };
+    assert_throws(new TypeError(), () => new WebAssembly.Global(argument, 0n));
+  }, `BigInt value for type ${type}`);
+}
+
+const valueArguments = [
+  [undefined, 0n],
+  [true, 1n],
+  [false, 0n],
```
Comment (xtuc) [inline test/js-api/global/constructor.any.js:130]: I wish that webassembly could assert types but it's correct

**Review by xtuc — APPROVED** [review]
Thanks

**Review by littledan — APPROVED** [review]
Thanks for these tests.

**Comment by littledan** [comment]
By the way, have these been tested against any implementation? You could try either V8 or SpiderMonkey

**Comment by lars-t-hansen** [comment]
BigInt integration hasn't landed yet in SpiderMonkey.

**Comment by Ms2ger** [comment]
They pass in V8.

**Pull request merged**